src/VisualStudio/Core/Def/Implementation/Workspace/SourceGeneratedFileManager.cs

```diff
@@ -0,0 +1,422 @@
+                    ErrorHandler.ThrowOnFailure(_runningDocumentTable.ModifyDocumentFlags(cookie, (uint)(_VSRDTFLAGS.RDT_CantSave | _VSRDTFLAGS.RDT_DontAddToMRU), fSet: 1));
+                }
+
+                if (windowFrame != null)
+                {
+                    openFile.SetWindowFrame(windowFrame);
+                }
+            }
+        }
+
+        void IRunningDocumentTableEventListener.OnCloseDocument(string moniker)
+        {
+            if (_openFiles.TryGetValue(moniker, out var openFile))
+            {
+                openFile.Dispose();
+                _openFiles.Remove(moniker);
+            }
+        }
+
+        void IRunningDocumentTableEventListener.OnRefreshDocumentContext(string moniker, IVsHierarchy hierarchy)
+        {
+        }
+
+        void IRunningDocumentTableEventListener.OnRenameDocument(string newMoniker, string oldMoniker, ITextBuffer textBuffer)
+        {
+        }
+
+        private class OpenSourceGeneratedFile : ForegroundThreadAffinitizedObject, IDisposable
+        {
+            private readonly SourceGeneratedFileManager _fileManager;
+            private readonly ITextBuffer _textBuffer;
+            private readonly Workspace _workspace;
+            private readonly ProjectId _projectId;
+            private readonly string _generatorTypeName;
+            private readonly string _generatedSourceHintName;
+
+            /// <summary>
+            /// A read-only region that we create across the entire file to prevent edits unless we are the one making them.
+            /// It's a dynamic read-only region that will allow edits if <see cref="_updatingBuffer"/> is set.
+            /// </summary>
+            private readonly IReadOnlyRegion _readOnlyRegion;
+            private bool _updatingBuffer = false;
+
+            /// <summary>
+            /// A cancellation token used for any background updating of this file; this is cancelled on the UI thread
+            /// when the file is closed.
+            /// </summary>
+            private readonly CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
+
+            /// <summary>
+            /// A queue used to batch updates to the file.
+            /// </summary>
+            private readonly AsyncBatchingDelay _batchingWorkQueue;
+
+            /// <summary>
+            /// The <see cref="IVsWindowFrame"/> of the active window. This may be null if we're in the middle of construction and
+            /// we haven't been given it yet.
+            /// </summary>
+            private IVsWindowFrame? _windowFrame;
+
+            private string? _windowFrameMessageToShow = null;
+            private ImageMoniker _windowFrameImageMonikerToShow = default;
+            private string? _currentWindowFrameMessage = null;
+            private ImageMoniker _currentWindowFrameImageMoniker = default;
+            private IVsInfoBarUIElement? _currentWindowFrameInfoBarElement = null;
```

Comment:
```suggestion
            private string? _windowFrameMessageToShow;
            private ImageMoniker _windowFrameImageMonikerToShow;
            private string? _currentWindowFrameMessage;
            private ImageMoniker _currentWindowFrameImageMoniker;
            private IVsInfoBarUIElement? _currentWindowFrameInfoBarElement;
```

Reply:
Going to leave it as is, since then it's explicit that's the starting state we're relying on.